### PR TITLE
12.1.Message Services, repository, controller, test

### PR DESCRIPTION
**Pull request by xalivan**

### CODE REVIEW

File: src/main/java/com/speaker/repository/MessageRepository.java [hunk 1]
```diff
@@ -3,8 +3,10 @@
 import com.speaker.entities.Message;
 
 import java.util.List;
+import java.util.Optional;
 
 public interface MessageRepository {
     List<Message> findAllByAccountId(int id);
-    int insert(Message message);
+
+    Optional<List<Integer>> insert(List<Message> message);
```
Comment (otkalenko) [inline src/main/java/com/speaker/repository/MessageRepository.java:11]: I think better named `createMessages`

File: src/main/java/com/speaker/repository/MessageRepositoryImpl.java [hunk 2]
```diff
@@ -26,10 +28,12 @@ public List<Message> findAllByAccountId(int id) {
...
-        return Objects.requireNonNull(dsl.insertInto(MESSAGE, MESSAGE.FROM_ACCOUNT_ID, MESSAGE.TO_ACCOUNT_ID, MESSAGE.DATE, MESSAGE.TEXT, MESSAGE.STATUS)
-                .values(message.getFromAccountId(), message.getToAccountId(), Timestamp.valueOf(LocalDateTime.now()), message.getText(), message.getStatus().toString())
-                .returningResult(MESSAGE.ID).fetchOne()).get(MESSAGE.ID);
+    public Optional<List<Integer>> insert(List<Message> message) {
+        return Optional.of(message.stream()
+                .map(ms -> Objects.requireNonNull(dsl.insertInto(MESSAGE, MESSAGE.FROM_ACCOUNT_ID, MESSAGE.TO_ACCOUNT_ID,
+                                MESSAGE.DATE, MESSAGE.TEXT, MESSAGE.STATUS)
+                        .values(ms.getFromAccountId(), ms.getToAccountId(),
+                                Timestamp.valueOf(LocalDateTime.now()), ms.getText(), ms.getStatus().toString())
+                        .returningResult(MESSAGE.ID).fetchOne()).get(MESSAGE.ID)).collect(Collectors.toList()));
```
Comment (otkalenko) [inline src/main/java/com/speaker/repository/MessageRepositoryImpl.java:37]: one execute per row in DB - is a bad way, if we want to create many rows in one operation need to use [batch operations](https://www.jooq.org/doc/latest/manual/sql-execution/batch-execution/)

File: src/main/java/com/speaker/service/MessageServiceImpl.java [hunk 3]
```diff
@@ -1,24 +1,51 @@
...
     @Override
     public List<Message> getMessagesByAccountId(int accountId) {
         return messageRepository.findAllByAccountId(accountId);
     }
 
     @Override
-    public int create(Message message) {
-        return messageRepository.insert(message);
+    public Response addMessage(List<MessageDTO> messageDTO) {
+        for (MessageDTO message : messageDTO) {
```
Comment (otkalenko) [inline src/main/java/com/speaker/service/MessageServiceImpl.java:31]: lets will use stream

File: src/main/java/com/speaker/service/util/StringParser.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package com.speaker.service.util;
+
+import java.util.Optional;
+
+import static java.util.Objects.nonNull;
+
+public class StringParser {
+    private static final Integer LIMIT_VALUE = 2;
+
+    public static Optional<Pair<String, String>> parseOptionalStringBySpace(String names) {
```
Comment (otkalenko) [inline src/main/java/com/speaker/service/util/StringParser.java:10]: do not see a need for this method

File: src/main/java/com/speaker/service/util/StringParser.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        if (nonNull(names)) {
+            return parser(names);
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<Pair<String, String>> parseOptionalStringBySpace(Optional<String> names) {
```
Comment (otkalenko) [inline src/main/java/com/speaker/service/util/StringParser.java:17]: the same

File: src/main/java/com/speaker/service/util/StringParser.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        if (nonNull(names) && names.isPresent()) {
+            return parser(names.get());
+        }
+        return Optional.empty();
+    }
+
+    private static Optional<Pair<String, String>> parser(String names) {
```
Comment (otkalenko) [inline src/main/java/com/speaker/service/util/StringParser.java:24]: splitBySpace

File: src/main/java/com/speaker/service/util/StringParser.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.util.Optional;
+
+import static java.util.Objects.nonNull;
+
+public class StringParser {
+    private static final Integer LIMIT_VALUE = 2;
+
+    public static Optional<Pair<String, String>> parseOptionalStringBySpace(String names) {
+        if (nonNull(names)) {
```
Comment (otkalenko) [inline src/main/java/com/speaker/service/util/StringParser.java:11]: move this in method `splitBySpace`

File: src/main/java/com/speaker/service/util/StringParser.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+            return parser(names);
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<Pair<String, String>> parseOptionalStringBySpace(Optional<String> names) {
+        if (nonNull(names) && names.isPresent()) {
```
Comment (otkalenko) [inline src/main/java/com/speaker/service/util/StringParser.java:18]: again `nonNull(names)` whay?

File: src/main/java/com/speaker/repository/MessageRepository.java (hunk 1 above)
Comment (xalivan) [reply src/main/java/com/speaker/repository/MessageRepository.java:11]: comments accepted, corrections made

File: src/main/java/com/speaker/repository/MessageRepositoryImpl.java (hunk 2 above)
Comment (xalivan) [reply src/main/java/com/speaker/repository/MessageRepositoryImpl.java:37]: comments accepted, corrections made. Added a list of messages in one query to the database

File: src/main/java/com/speaker/repository/MessageRepository.java (hunk 1 above)
Comment (xalivan) [reply src/main/java/com/speaker/repository/MessageRepository.java:11]: Comments accepted, corrections made. Method name changed

File: src/main/java/com/speaker/repository/MessageRepositoryImpl.java (hunk 2 above)
Comment (xalivan) [reply src/main/java/com/speaker/repository/MessageRepositoryImpl.java:37]: Comments accepted, corrections made. Add batch function and list of query

File: src/main/java/com/speaker/service/MessageServiceImpl.java (hunk 3 above)
Comment (xalivan) [reply src/main/java/com/speaker/service/MessageServiceImpl.java:31]: Comments accepted, corrections made. Thank you

File: src/main/java/com/speaker/service/util/StringParser.java (hunk 4 above)
Comment (xalivan) [reply src/main/java/com/speaker/service/util/StringParser.java:10]: Comments accepted, corrections made. Method has been deleted

File: src/main/java/com/speaker/service/util/StringParser.java (hunk 7 above)
Comment (xalivan) [reply src/main/java/com/speaker/service/util/StringParser.java:11]: Comments accepted, corrections made. Name of method has been changed

File: src/main/java/com/speaker/service/util/StringParser.java (hunk 5 above)
Comment (xalivan) [reply src/main/java/com/speaker/service/util/StringParser.java:17]: Name of method has been changed

File: src/main/java/com/speaker/service/util/StringParser.java (hunk 8 above)
Comment (xalivan) [reply src/main/java/com/speaker/service/util/StringParser.java:18]: Comments accepted, corrections made. thank you. it is mistake

File: src/main/java/com/speaker/service/util/StringParser.java (hunk 6 above)
Comment (xalivan) [reply src/main/java/com/speaker/service/util/StringParser.java:24]: Comments accepted, corrections made. changed

File: src/test/java/com/speaker/service/MessageServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        verify(messageConvertor, never()).convertToMessage(messageDTO.get(INDEX_1), null, FRIEND_ID);
+        verify(messageRepository, never()).createMessages(message);
+    }
+
+    @Test
+    void messageNotAddingWhenAccountNameIsNull() {
+        List<MessageDTO> messageDTO = generateMessageDTOList(null, FRIEND_FIRST_LAST_NAME);
+        List<Message> message = generateMessageList();
+        assertThat(messageService.addMessage(messageDTO), is(Response.FALSE));
+        verify(accountRepository, never()).findAccountIdByNameAndLastName(null, null);
```
Comment (otkalenko) [inline src/test/java/com/speaker/service/MessageServiceImplTest.java:83]: I think this is excessive

File: src/test/java/com/speaker/service/MessageServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+        verify(accountRepository, times(2)).findAccountIdByNameAndLastName(FRIEND_FIRST_NAME, LAST_NAME);
+        verify(messageConvertor, never()).convertToMessage(messageDTO.get(INDEX_0), null, FRIEND_ID);
```
Comment (otkalenko) [inline src/test/java/com/speaker/service/MessageServiceImplTest.java:85]: the same

File: src/test/java/com/speaker/service/MessageServiceImplTest.java [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+        verify(messageConvertor, never()).convertToMessage(messageDTO.get(INDEX_1), null, FRIEND_ID);
```
Comment (otkalenko) [inline src/test/java/com/speaker/service/MessageServiceImplTest.java:86]: also

File: src/test/java/com/speaker/service/MessageServiceImplTest.java [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+        verify(messageRepository, never()).createMessages(message);
```
Comment (otkalenko) [inline src/test/java/com/speaker/service/MessageServiceImplTest.java:87]: also

File: src/test/java/com/speaker/service/MessageServiceImplTest.java [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+        assertThat(messageService.addMessage(messageDTO), is(Response.FALSE));
+        verify(accountRepository, times(2)).findAccountIdByNameAndLastName(ACCOUNT_FIRST_NAME, LAST_NAME);
+        verify(accountRepository, times(2)).findAccountIdByNameAndLastName(FRIEND_FIRST_NAME, LAST_NAME);
+        verify(messageConvertor, never()).convertToMessage(messageDTO.get(INDEX_0), null, FRIEND_ID);
+        verify(messageConvertor, never()).convertToMessage(messageDTO.get(INDEX_1), null, FRIEND_ID);
+        verify(messageRepository, never()).createMessages(message);
+    }
+
+    @Test
+    void messageNotAddingWhenAccountNameIsNull() {
```
Comment (otkalenko) [inline src/test/java/com/speaker/service/MessageServiceImplTest.java:79]: please add an access modifier

Comment (otkalenko) [reply src/test/java/com/speaker/service/MessageServiceImplTest.java:79]: messageNotCreatedWhenAccountNameIsNull

File: src/test/java/com/speaker/service/MessageServiceImplTest.java (hunk 9 above)
Comment (xalivan) [reply src/test/java/com/speaker/service/MessageServiceImplTest.java:83]: Comments accepted, corrections made.  "Verify" has been deleted

File: src/test/java/com/speaker/service/MessageServiceImplTest.java (hunk 10 above)
Comment (xalivan) [reply src/test/java/com/speaker/service/MessageServiceImplTest.java:85]: Comments accepted, corrections made.  "Verify" has been deleted

File: src/test/java/com/speaker/service/MessageServiceImplTest.java (hunk 11 above)
Comment (xalivan) [reply src/test/java/com/speaker/service/MessageServiceImplTest.java:86]: Comments accepted, corrections made.  "Verify" has been deleted

File: src/test/java/com/speaker/service/MessageServiceImplTest.java (hunk 13 above)
Comment (xalivan) [reply src/test/java/com/speaker/service/MessageServiceImplTest.java:76]: Comments accepted, corrections made.

**Pull request merged**